File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hashmap = {}
        results = []
        
        for num in nums:
            if num not in hashmap.keys():
                hashmap[num] = 1
            else:
                hashmap[num] += 1
        
        for key, value in hashmap.items():
            results.append([value, key])
        results.sort()

        r = []

        while len(r) < k:
            r.append(results.pop()[1])
        return r

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();

        for (Integer num: nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }

        System.out.println(map.toString());

        List<int[]> list = new ArrayList<>();

        for (Map.Entry<Integer, Integer> entry: map.entrySet()) {
            list.add(new int[] {entry.getValue(), entry.getKey()});
        }

        list.sort((a,b) -> b[0] - a[0]);
        
        int[] res = new int[k];

        for(int i = 0; i < res.length; i++) {
            res[i] = list.get(i)[1];
        }

        return res;
    }
}